tools/crd-bumper/README.md

```diff
@@ -0,0 +1,117 @@
+# CRD Version Bumper
+
+Bump the CRD version, adding conversion webhooks and tests.
+
+See Kubebuilder's [Tutorial: Multi-Version API](https://book.kubebuilder.io/multiversion-tutorial/tutorial) for a description of the mechanism. For more detail read the Kubernetes document [Versions in CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/).
+
+## Hub and Spokes
+
+This tool implements the hub-and-spoke model of CRD versioning. See Kubebuilder's [Hubs, spokes, and other wheel metaphors](https://book.kubebuilder.io/multiversion-tutorial/conversion-concepts) for a description of the model.
+
+The new CRD version will be the new hub, and the previous hub will become a new spoke. (Note: in the Kubebuilder book example, the old version is the hub and never changes, and all new versions are spokes.)
```

Comment:
So is the Kubebuilder book/model incorrect?

Reply:
It's a very simple example that is easy to explain and show. But it's not a practical model.

It has a few down-sides. Here are a few I can rattle off immediately:
1. it means you can never deprecate and remove that first version of the API. Or at least, it never explains how one would accomplish that.
2. it means that first version of the resource is the version that is stored in etcd, forever. It doesn't explain how newer versions will be stored.
3. it doesn't include a loss-less conversion model. It simply says, "go implement that, here are the expected function names of the API."

I experimented with the version described in the kubebuilder book 3-4 years ago and it felt like the old "paint yourself into a corner" kind of scenario. So I put it aside. Then about a year and a half ago I stumbled on the cluster-api repository and saw a more enlightened way to do this.

Reply:
Thanks for breaking that down. I need to go check this stuff out more.